### PR TITLE
[software-bridges 2/x] report VF representor name in SriovNetworkNodeState.Status field

### PR DESCRIPTION
**Pull request by ykulazhenkov**
Add support for reporting VF representor name when the NIC is in switchev mode

Report VF representor field as described in the [desing doc](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/doc/design/software-bridge-management.md)

### CODE REVIEW

File: pkg/host/internal/sriov/sriov_test.go [hunk 1]
```diff
@@ -38,10 +40,12 @@ var _ = Describe("SRIOV", func() {
 		testCtrl = gomock.NewController(GinkgoT())
 		netlinkLibMock = netlinkMockPkg.NewMockNetlinkLib(testCtrl)
 		dputilsLibMock = dputilsMockPkg.NewMockDPUtilsLib(testCtrl)
+		sriovnetLibMock = sriovnetMockPkg.NewMockSriovnetLib(testCtrl)
```
Comment (SchSeba) [inline pkg/host/internal/sriov/sriov_test.go:43]: can we add a unit-test to validate the vf representor return?

File: go.mod [hunk 2]
```diff
@@ -4,6 +4,7 @@ go 1.21
 
 require (
 	github.com/Masterminds/sprig/v3 v3.2.2
+	github.com/Mellanox/sriovnet v1.0.3
 	github.com/blang/semver v3.5.1+incompatible
```
Comment (SchSeba) [inline go.mod:8]: nit: please open an issue in the operator repo I would like to see this package moving under the npwg org if possible

**Review by SchSeba — COMMENTED** [review]
great work, please just implement a small unit test to validate the return of the representor

File: pkg/host/internal/sriov/sriov_test.go (hunk 1 above)
Comment (ykulazhenkov) [reply pkg/host/internal/sriov/sriov_test.go:47]: I added one simple test for Discover function (positive scenario only)

File: go.mod (hunk 2 above)
Comment (ykulazhenkov) [reply go.mod]: It is already part of the NPWG. I updated code to use the right version

**Review by adrianchiris — APPROVED** [review]

**Comment by ykulazhenkov** [comment]
I rebased the PR and applied requested changes to the code. @SchSeba  please take another look, thx

**Review by SchSeba — APPROVED** [review]
LGTM

/hold
waiting for the sriov operator release

**Review by ykulazhenkov — COMMENTED** [review]
@SchSeba @adrianchiris The new version of the operator has been released. Can we merge this PR? thx

**Review by adrianchiris — APPROVED** [review]

**Pull request merged**